### ISSUE TITLE
Rename column operation being ran against the wrong database when using multiple DbContexts

### ISSUE THREAD

**Issue by robert396** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#393]
### Steps to reproduce
[source example](https://github.com/robert396/multipledbcontextissue)

### The issue
When using multiple DbContext's targeting different databases, the RenameColumn operation is executed using the name of the first database as the schema which causes the `GetCreateTable` method of the `MySqlMigrationsSqlGenerator` to fail when migrating via code.

First DbContext: `ClientDbContext` -> `sample_clients` 
Second DbContext: `SiteDbContext` -> `sample_sites`
Table column to rename: `SiteDomain` to `Domain`
Method ran: `siteDbContext.Database.Migrate()`

```
Exception message: Table 'sample_clients.sites' doesn't exist
Stack trace:
   at MySql.Data.MySqlClient.MySqlDataReader.ActivateResultSet(ResultSet resultSet)
   at MySql.Data.MySqlClient.MySqlDataReader.<ReadFirstResultSetAsync>d__64.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MySql.Data.MySqlClient.MySqlDataReader.<CreateAsync>d__63.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MySql.Data.MySqlClient.CommandExecutors.TextCommandExecutor.<ExecuteReaderAsync>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Internal.MySqlOptions.ExecuteCreateTable(DbConnection connection, ISqlGenerationHelper sqlGenerationHelper, String table, String schema)
   at Microsoft.EntityFrameworkCore.Internal.MySqlOptions.GetCreateTable(String connectionString, ISqlGenerationHelper sqlGenerationHelper, String table, String schema)
   at Microsoft.EntityFrameworkCore.Migrations.MySqlMigrationsSqlGenerator.Generate(RenameColumnOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.<>c.<.cctor>b__63_20(MigrationsSqlGenerator g, MigrationOperation o, IModel m, MigrationCommandListBuilder b)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(IReadOnlyList`1 operations, IModel model)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.GenerateUpSql(Migration migration)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.<>c__DisplayClass13_2.<GetMigrationCommandLists>b__2()
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.Migrate(DatabaseFacade databaseFacade)
   at MultipleDbIssues.Startup.Configure(IApplicationBuilder app, IHostingEnvironment env) in D:\Workspaces\DotNetCore\MultipleDbIssues\MultipleDbIssues\Startup.cs:line 58
```

The RenameColumnOperation should be run against `sample_sites.sites` instead of `sample_clients`

### Current Workaround

Currently to get around this issue I am having to use a custom service provider on the `SiteDbContext` to use an internal service provider that just registers all of the required services via the `AddEntityFrameworkMySql` method, and then replace the current registered provider for `IMigrationsSqlGenerator` with a dummy class that just inherits from the `MySqlMigrationsSqlGenerator` class as a base, which seems to fix the issue as the sql generator is now using the correct connection string, instead of the connection string for the first DbContext.

### Further technical details

MySQL version: 5.7.19-log
Operating system: Windows 10
Pomelo.EntityFrameworkCore.MySql version: 2.0.0

**Comment by caleblloyd** [comment]
Have you tried this against another provider such as SQL Server?

I looked at all of the code that touches [RenameColumnOperation](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/101e9ddbb14dc4f9fe7de634db91ef40ea5fdf6b/src/EFCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs#L338-L370) and all of the code that touches [GetCreateTable](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/fe78d7240faac42232471de46ae1475bdbab0d03/src/EFCore.MySql/Internal/MySqlOptions.cs#L46-L93).  It looks like we just use the `operation.Schema` that is passed from upstream Entity Framework in the `Generate` function that handles `RenameColumnOperation` so I'm thinking the issue may be upstream.  I could have missed something though.

What commands are you using to migrate your database up?

**Comment by robert396** [comment]
@caleblloyd The issue seems to stem from the `MySqlOptions` where it is being used in a singleton context, when the `_relationalOptions` property is being set it contains the connection string for the first `DbContext` that is used in the application, and because it is never updated it means that when it is running:
```
private static string GetCreateTable(string connectionString, ISqlGenerationHelper sqlGenerationHelper, string table, string schema)
{
	using (var connection = new MySqlConnection(connectionString))
	{
		connection.Open();
		return ExecuteCreateTable(connection, sqlGenerationHelper, table, schema);
	}
}
```
it is using the incorrect connection string for the current DbContext, the only way I can get around this issue at the moment is by creating a custom service collection and have the 2 DbContext's use seperate service providers.

I also tried the same migrations using `SqlServer` and that worked fine see [here](https://github.com/robert396/multipledbcontextissue/tree/sqlserver) for the SqlServer branch.

**Comment by johnnynibbles** [comment]
I have recently come across this same problem after upgrading from 1.1 to 2.0.  I was able to have  multiple context migrating without issue before.  I have created a repo with an example that runs it in containers, but sometimes the first run will error during connection because of imperfect healthcheck on MySQL server.  If you run it a second time it will be able to connect.[Error Example Repo](https://github.com/johnnynibbles/PomeloMySqlError)

I believe that @robert396 is correct that whatever is loading the _relationalOptions is only happening once.
